CHANGELOG.md

```diff
@@ -41,6 +41,7 @@ Copy and pasting the git commit messages is __NOT__ enough.
 - `SumoTrafficSimulation` no longer reports that it manages vehicles when it is disconnected.
 - Fixed waypoints so that they will always be the length of the `lookahead` parameter, even in junctions.
 - Fixed an issue where a single waypoint would appear in off-route lanes for missions with a goal.
+- Fixed an issue where the `off_route` event would be incorrectly triggered when driving on overlapping lanes.
```

Comment:
What I will say is that this does not solve the problem fully.

The other issue at the core is that we have overpass lanes overlapping on the main road without being separated by height layer, and without being listed as a junction. This means that just driving through the intersection between the overpass and the other highway can result in ambiguous road association.